corehq/apps/users/models.py

```diff
@@ -1610,8 +1596,8 @@ def _get_viewable_report_slugs(self, domain):
             else:
                 return role.permissions.view_report_list
         else:
-            dm = self.get_domain_membership(domain)
-            return dm.viewable_reports() if dm else []
+            membership = self.get_domain_membership(domain) or self.get_enterprise_membership(domain)
+            return membership.viewable_reports() or []
```

Comment:
can `membership` be `None`. it looks like the old code was guarding against that, and this no longer does

Reply:
Not following. If `membership` is `None`, this returns an empty list, in both the old code (1597) and the new code (1562). Am I missing something?

Reply:
if `membership` is `None`, wont `membership.viewable_reports()` throw an `AttributeError`

Reply:
offlined, resolved